### ISSUE TITLE
Provider constraint <6.0.0 causes conflicts in other modules

### ISSUE THREAD

**Issue by prshoper** [terraform-google-modules/terraform-google-bastion-host#217]
### TL;DR

Using a Terraform provider version constraint <6.0.0 causes a conflict or bug in other modules. The issue likely occurs due to incompatibility between this version constraint and the rest of the configuration. Updating or adjusting the provider version might be required to resolve the conflict.

### Expected behavior

Terraform should execute the plan and apply the configuration without issues, respecting the provider version constraint <=6.0.0 and maintaining compatibility with all modules

### Observed behavior

The provider version constraint <=6.0.0 generates a conflict or bug in other modules, leading to a failure. The error message communicates that no successful version of the provider could be used due to the constraint.

### Terraform Configuration

```hcl
module "mysql-db" {
  source                          = "GoogleCloudPlatform/sql-db/google//modules/mysql"
  version                         = "22.1.0"
...
}

AND
module "iap_bastion" {
  source  = "terraform-google-modules/bastion-host/google"
  version = "7.1.0"
....
}
```

### Terraform Version

```sh
Terraform v1.9.8
```
